packages/augur-sdk/src/state/db/MarketDB.ts

```diff
@@ -0,0 +1,112 @@
+import * as _ from "lodash";
+import { DerivedDB } from "./DerivedDB";
+import { DB } from "./DB";
+import { toAscii } from "../utils/utils";
+
+// Need this interface to access these items on the documents
+interface MarketDataDoc extends PouchDB.Core.ExistingDocument<PouchDB.Core.AllDocsMeta> {
+  market: string;
+  topic: string;
+  extraInfo: string;
+}
+
+// because flexsearch is a UMD type lib
+import FlexSearch = require("flexsearch");
+
+/**
+ * Market specific derived DB intended for filtering purposes
+ */
+export class MarketDB extends DerivedDB {
+  private flexSearch?: FlexSearch;
+
+  constructor(db: DB, networkId: number) {
+    super(db, networkId, "Markets", ["MarketCreated", "MarketVolumeChanged", "MarketOIChanged"], ["market"]);
+    // TODO Register callback after CurrentOrders is updated during blockstream. Use augurEmitter to subscribe to it and do post processing on the orderbook.
+
+    this.flexSearch = new FlexSearch({
+      doc: {
+        id: "id",
+        start: "start",
+        end: "end",
+        field: [
+          "market",
+          "topic",
```

Comment:
just a node: We'll need to refactor when topic is removed and 2-3 categories is added via extraInfo